src/foundation.cs

```diff
@@ -10351,6 +10351,18 @@ partial interface NSItemProvider : NSCopying {
 		[MacCatalyst (13, 0)]
 		[Export ("preferredPresentationStyle", ArgumentSemantic.Assign)]
 		UIPreferredPresentationStyle PreferredPresentationStyle { get; set; }
+
+		// extension methods from CloudKit
+
+		[Async]
```

Comment:
This async looks like may be troublesome, it is not a completion handler but a preparation handler we do not know when this is called so I would remove the Async here

```suggestion
```

Reply:
Agreed, does look like it will be execute before anything happens, that means that if we callout async we would have a completely wrong behaviour.